shell/platform/android/io/flutter/embedding/engine/systemchannels/SpellCheckChannel.java

```diff
@@ -0,0 +1,81 @@
+package io.flutter.embedding.engine.systemchannels;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Log;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.JSONMethodCodec;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.ArrayList;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+public class SpellCheckChannel {
+  private static final String TAG = "SpellCheckChannel";
+
+  @NonNull public final MethodChannel channel;
+  @Nullable private SpellCheckMethodHandler spellCheckMethodHandler;
+
+  @NonNull @VisibleForTesting
+  final MethodChannel.MethodCallHandler parsingMethodHandler =
+      new MethodChannel.MethodCallHandler() {
+        @Override
+        public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
+          if (spellCheckMethodHandler == null) {
+            // If no explicit SpellCheckMethodHandler has been registered then we don't
+            // need to forward this call to an API. Return.
+            return;
+          }
+          String method = call.method;
+          Object args = call.arguments;
+          Log.v(TAG, "Received '" + method + "' message.");
+          switch (method) {
+            case "SpellCheck.initiateSpellCheck":
+              try {
+                final JSONArray argumentList = (JSONArray) args;
+                String locale = argumentList.getString(0);
+                String text = argumentList.getString(1);
+                spellCheckMethodHandler.initiateSpellCheck(locale, text);
+                result.success(null);
+              } catch (JSONException exception) {
+                result.error("error", exception.getMessage(), null);
+              }
+              break;
+            default:
+              result.notImplemented();
+              break;
+          }
+        }
+      };
+
+  public SpellCheckChannel(@NonNull DartExecutor dartExecutor) {
+    this.channel = new MethodChannel(dartExecutor, "flutter/spellcheck", JSONMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(parsingMethodHandler);
+  }
+
+  /** Responsible for sending spell check results through this channel. */
```

Comment:
some details about what data flows through the channel in a diagram would be helpful.  Since this would be higher level than this specific method, consider adding it to the javadoc of this class.

Reply:
I added more details to the javadoc of this class, but in terms of adding a diagram, do you have any examples?